mlflow/genai/labeling/labeling.py

```diff
@@ -143,7 +175,39 @@ def add_traces(
         Returns:
             LabelingSession: The updated labeling session.
         """
-        return LabelingSession(self._session.add_traces(traces))
+        import pandas as pd
+
+        from mlflow.exceptions import MlflowException
+        from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+        if isinstance(traces, pd.DataFrame):
+            if "trace" not in traces.columns:
+                raise MlflowException(
+                    "traces must have a 'trace' column like the result of mlflow.search_traces()",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+            traces = traces["trace"].to_list()
+
+        trace_list: list[Trace] = []
+        for trace in traces:
+            if isinstance(trace, str):
+                trace_list.append(Trace.from_json(trace))
+            elif isinstance(trace, Trace):
+                trace_list.append(trace)
+            elif trace is None:
+                raise MlflowException(
+                    "trace cannot be None. Must be mlflow.entities.Trace or its json string "
+                    "representation.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+            else:
+                raise MlflowException(
+                    f"Expected mlflow.entities.Trace or json string, got {type(trace).__name__}",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
```

Comment:
For reference, see the implementation of LabelingSession.add_traces in `databricks.agents`